src/views/pages/profiles/import-profile/ImportProfileTs.ts

```diff
@@ -142,6 +158,46 @@ export default class ImportProfileTs extends Vue {
         this.initialized = true;
     }
 
+    /**
+     * Fetch account balances and map to address
+     * @return {void}
+     */
+    @Watch('optInSelectedAccounts')
+    private async initOptInAccounts() {
+        if (this.optInInitialized) {
+            return;
+        }
+
+        // - generate addresses
+        const possibleOptInAccounts = this.accountService.generateAccountsFromMnemonic(
+            new MnemonicPassPhrase(this.currentMnemonic),
+            this.currentProfile.networkType,
+            10,
+            Network.BITCOIN,
+        );
+
+        // whitelist opt in accounts
+        const key = this.currentProfile.networkType === NetworkType.MAIN_NET ? 'MAINNET' : 'TESTNET';
+        const whitelisted = process.env[`VUE_APP_${key}_WHITELIST`] ? process.env[`VUE_APP_${key}_WHITELIST`].split(',') : [];
+        const optInAccounts = possibleOptInAccounts.filter((account) => whitelisted.indexOf(account.publicKey) >= 0);
+        if (optInAccounts.length === 0) {
+            return;
+        }
+        this.optInAddressesList = optInAccounts.map((account) => account.address);
+
+        // fetch accounts info
+        const repositoryFactory = this.$store.getters['network/repositoryFactory'] as RepositoryFactory;
+        const accountsInfo = await repositoryFactory.createAccountRepository().getAccountsInfo(this.addressesList).toPromise();
```

Comment:
should we check `repositoryFactory` is not null or undefined here to support the offline mode?